crates/wkg/src/oci.rs

```diff
@@ -129,6 +130,24 @@ pub struct PushArgs {
 
     /// The path to the file to push
     pub file: PathBuf,
+
+    /// Add an OCI annotation to the image manifest
+    #[clap(long = "annotation", value_parser = parse_key_val::<String, String>)]
+    pub annotation: Vec<(String, String)>,
+}
+
+/// Parse a single key-value pair
+fn parse_key_val<T, U>(s: &str) -> Result<(T, U), Box<dyn Error + Send + Sync + 'static>>
+where
+    T: std::str::FromStr,
+    T::Err: Error + Send + Sync + 'static,
+    U: std::str::FromStr,
+    U::Err: Error + Send + Sync + 'static,
+{
+    let pos = s
```

Comment:
Nit: I think it would be more readable to use `s.split_once` and match on the option instead. I know you can technically split the string and take ownership, but this is a CLI, so to_owned on them should be just fine